src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectNative.cs

```diff
@@ -35,9 +33,9 @@ internal class TdsParserStateObjectNative : TdsParserStateObject
 
         internal override uint Status => _sessionHandle != null ? _sessionHandle.Status : TdsEnums.SNI_UNINITIALIZED;
 
-        internal override object SessionHandle => _sessionHandle;
+        internal override SessionHandle SessionHandle => SessionHandle.FromNativeHandle(_sessionHandle);
 
-        protected override object EmptyReadPacket => s_cachedEmptyReadPacketObjectPointer;
+        protected override PacketHandle EmptyReadPacket => default;
```

Comment:
This Property accessor could be moved into the Base class now since it's no longer an `object` but a common concrete type. You could take care of it in a PR after this one is merged.

Reply:
There are only two uses of the property, one of these I've already removed and have mis-merged back in `IsConnectionAlive` and the other in `SNIWritePacket` where it's only used as a default value so I should be able to remove it entirely.